hugo/content/docs/smodel.md

```diff
@@ -0,0 +1,554 @@
+
+### SConnectableElementImpl
+
+A connectable element is on that can have outgoing and incoming edges. It can be the source or target element of an edge. There are two kinds of connectable elements: nodes and ports.
+
+*Properties:*
+
+* `strokeWidth: number`: The stroke width of the element. Defaults to `0`.
+
+*Inheritance:*
+
+[`SShapeElementImpl`](#sshapeelementimpl) &rarr; [`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+### SShapeElementImpl
+
+Abstract class for elements with a position and size.
+
+*Properties:*
+
+* `position: Point`: The position of the element. Defaults to `{x: 0, y: 0}`.
+* `size: Dimension`: The size of the element. Defaults to `{width: -1, height: -1}`.
+* `layoutOptions: ModelLayoutOptions` - *optional*: Options for the layout of the element.
+
+*Inheritance:*
+
+[`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+### SRoutableElementImpl
+
+Abstract class for edges.
+
+*Properties:*
+
+* `routerKind: string` - *optional*: The kind of router to use for the edge.
+* `routingPoints: Point[]` - *optional*: The routing points of the edge. Defaults to an empty array.
+* `sourceId: string`: The id of the source element.
+* `targetId: string`: The id of the target element.
+* `sourceAnchorCorrection: number` - *optional*: The correction of the source anchor. Defaults to `0`.
+* `targetAnchorCorrection: number` - *optional*: The correction of the target anchor. Defaults to `0`.
+
+*Inheritance:*
+
+[`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+## Classes Inheritance
+
+{{< mermaid class="text-center">}}
+flowchart TD;
+SModelElementImpl
+SParentElementImpl
+SChildElementImpl
+SModelRootImpl
+ModelIndexImpl
+SGraphImpl
+ViewportRootElement
+SNodeImpl
+SConnectableElementImpl
+SShapeElementImpl
+SPortImpl
+SEdgeImpl
+SRoutableElementImpl
+SLabelImpl
+SCompartmentImpl
+SGraphIndex
+
+SParentElementImpl --> SModelElementImpl
+SChildElementImpl --> SParentElementImpl
+SModelRootImpl --> SParentElementImpl
+SGraphImpl --> ViewportRootElement
+ViewportRootElement --> SModelRootImpl
+SNodeImpl --> SConnectableElementImpl
+SConnectableElementImpl --> SShapeElementImpl
+SShapeElementImpl --> SChildElementImpl
+SPortImpl --> SConnectableElementImpl
+SEdgeImpl --> SRoutableElementImpl
+SRoutableElementImpl --> SChildElementImpl
+SLabelImpl --> SShapeElementImpl
+SCompartmentImpl --> SShapeElementImpl
+SGraphIndex --> ModelIndexImpl
+{{< /mermaid>}}
+
+## Features
+
+Features can be viewed as a set of functionalities that a given element can support. They can enrich the behavior and usability of the diagram model. SModelElements can have default features that will apply to all model elements that are instance of this SModelElement.
```

Comment:
Features can be viewed as a set of functionalities that a given element can support. They can enrich the behavior and usability of the diagram model. [Features serve as light-weight extension mechanism in Sprotty to add additional functionalities in upcoming Sprotty versions or by users of Sprotty (?).]

[line break]

SModelElements can have default features that will apply to all model elements that are instance of this SModelElement.[ Default f]eatures are specified in the `DEFAULT_FEATURES` field, which could be overridden in inherited classes. This way, some default features will be enabled for all instances of the class.

Let's have a look at the `SNodeImpl` class and how it implements default features: